connectors/spark/src/main/scala/io/unitycatalog/spark/token/UCTokenProvider.java

```diff
@@ -0,0 +1,70 @@
+package io.unitycatalog.spark.token;
+
+import static org.sparkproject.guava.base.Preconditions.checkArgument;
+
+import io.unitycatalog.spark.UCHadoopConf;
+import io.unitycatalog.spark.utils.OptionsUtil;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+
+public interface UCTokenProvider {
+  /**
+   * Returns the access token for Unity Catalog authentication.
+   *
+   * @return the access token string
+   */
+  String accessToken();
+
+  /**
+   * Returns the configuration properties associated with this token provider.
+   *
+   * @return a map of configuration key-value pairs
+   */
+  Map<String, String> properties();
+
+  /**
+   * Creates a token provider from configuration options. Returns {@link FixedUCTokenProvider} if
+   * a token is provided, or {@link OAuthUCTokenProvider} if OAuth credentials are provided.
+   *
+   * @param options containing authentication credentials
+   * @return a token provider instance
+   * @throws IllegalArgumentException if options are missing or incomplete
+   */
+  static UCTokenProvider create(Map<String, String> options) {
+    // If token is available, use FixedUCTokenProvider.
+    String token = options.get(OptionsUtil.TOKEN);
+    if (token != null) {
+      return new FixedUCTokenProvider(token);
+    }
+
+    // If OAuth options is available, use OAuthUCTokenProvider.
+    String oauthUri = options.get(OptionsUtil.OAUTH_URI);
+    String oauthClientId = options.get(OptionsUtil.OAUTH_CLIENT_ID);
+    String oauthClientSecret = options.get(OptionsUtil.OAUTH_CLIENT_SECRET);
+    if (oauthUri != null || oauthClientId != null || oauthClientSecret != null) {
+      checkArgument(oauthUri != null && oauthClientId != null && oauthClientSecret != null,
+          "Incomplete OAuth configuration detected. All of the keys are required: " +
+              "spark.sql.catalog.<catalogName>.oauthUri, " +
+              "spark.sql.catalog.<catalogName>.oauthClientId, " +
+              "spark.sql.catalog.<catalogName>.oauthClientSecret. " +
+              "Please ensure they are all set.");
```

Comment:
this error message does not make sense when this method has been called from `UCTokenProvider create(Configuration conf)` where the problem is not user not setting these confs, but somehow hadoop conf being incorrect. 


so the error message needs to be different based on the caller of this generic method